scripts/leapp_preupgrade.py

```diff
@@ -0,0 +1,340 @@
+import json
+import os
+import subprocess
+
+
+# Both classes taken from:
+# https://github.com/oamg/convert2rhel-worker-scripts/blob/main/scripts/preconversion_assessment_script.py
+class ProcessError(Exception):
+    """Custom exception to report errors during setup and run of leapp"""
+
+    def __init__(self, message):
+        super(ProcessError, self).__init__(message)
+        self.message = message
+
+
+class OutputCollector(object):
+    """Wrapper class for script expected stdout"""
+
+    def __init__(self, status="", message="", report="", entries=None):
+        self.status = status
+        self.message = message
+        self.report = report
+        self.tasks_format_version = "1.0"
+        self.tasks_format_id = "oamg-format"
+        self.entries = entries
+        self.report_json = None
+
+    def to_dict(self):
+        # If we have entries, then we change report_json to be a dictionary
+        # with the needed values, otherwise, we leave it as `None` to be
+        # transformed to `null` in json.
+        if self.entries:
+            self.report_json = {
+                "tasks_format_version": self.tasks_format_version,
+                "tasks_format_id": self.tasks_format_id,
+                "entries": self.entries,
+            }
+
+        return {
+            "status": self.status,
+            "message": self.message,
+            "report": self.report,
+            "report_json": self.report_json,
+        }
+
+
+def get_rhel_version():
+    """Currently we execute the task only for RHEL 7 or 8"""
+    print("Checking OS distribution and version ID ...")
+    try:
+        distribution_id = None
+        version_id = None
+        with open("/etc/os-release", "r") as os_release_file:
+            for line in os_release_file:
+                if line.startswith("ID="):
+                    distribution_id = line.split("=")[1].strip().strip('"')
+                elif line.startswith("VERSION_ID="):
+                    version_id = line.split("=")[1].strip().strip('"')
+            return distribution_id, version_id
+    except IOError:
+        print("Couldn't read /etc/os-release")
+    return None, None
+
+
+def exit_for_non_eligible_releases(release):
+    print("Exit if not RHEL 7 or RHEL 8 ...")
+    major_version, _ = release.split(".") if release is not None else (None, None)
+    return release is None or major_version not in ["7", "8"]
```

Comment:
as said in other comments, since we don't know how we will handle it on insights side I woul keep both preupgrades with same functionality and drop it in future if we have to. 

Insights doesn't have any filter for systems that can be selected fro execution, as far as I know you can do it for any system right now that is correctly connected with subscription manager, rhc and insights-client, either it will fail or not.

Only remediations in advisor are limited to be shown on supported upgrade paths

Reply:
Alright, so does it make sense to us to exit the task if it is not in supported upgrade paths? Some endpoint would be really useful across our uses :D

Reply:
well the ansible task also works only for rhel 7 and rhel 8, but doesn't care about specific minor versions, that is why the exit is also in here

Reply:
alright, I guess we can talk about possible improvements for both of the scripts in other issue